### ISSUE TITLE
1.1.6 doesn't upgrade to 2.0.0-dev well

### ISSUE THREAD

**Issue by fnoop** [goodrobots/maverick#783]
```
Notice: /Stage[main]/Openldap::Client::Config/Augeas[ldap.conf]/returns: executed successfully
Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install slapd' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
slapd is already the newest version (2.4.42+dfsg-2ubuntu3.3).
The following packages were automatically installed and are no longer required:
  grub-pc-bin libllvm5.0 libproj-dev libvtk-java libvtk5.10
  linux-headers-4.4.0-109 linux-headers-4.4.0-109-generic
  linux-headers-4.4.0-112 linux-headers-4.4.0-112-generic
  linux-headers-4.4.0-116 linux-headers-4.4.0-116-generic
  linux-headers-4.4.0-124 linux-headers-4.4.0-124-generic
  linux-image-4.4.0-109-generic linux-image-4.4.0-112-generic
  linux-image-4.4.0-116-generic linux-image-4.4.0-124-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up slapd (2.4.42+dfsg-2ubuntu3.3) ...
  Moving old database directory to /var/backups:
  - directory unknown... done.
  Creating initial configuration... done.
  Creating LDAP directory... done.
Job for slapd.service failed because the control process exited with error code. See "systemctl status slapd.service" and "journalctl -xe" for details.
invoke-rc.d: initscript slapd, action "start" failed.
* slapd.service - LSB: OpenLDAP standalone server (Lightweight Directory Access Protocol)
   Loaded: loaded (/etc/init.d/slapd; bad; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2018-07-06 22:43:48 BST; 8ms ago
     Docs: man:systemd-sysv-generator(8)
  Process: 31496 ExecStart=/etc/init.d/slapd start (code=exited, status=1/FAILURE)

Jul 06 22:43:48 goodrobots-maverick slapd[31503]: ldif_read_file: Permission ...
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: DIGEST-MD5 common mech free
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: DIGEST-MD5 common mech free
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: slapd stopped.
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: connections_destroy: nothin...
Jul 06 22:43:48 goodrobots-maverick slapd[31496]:    ...fail!
Jul 06 22:43:48 goodrobots-maverick systemd[1]: slapd.service: Control proce...1
Jul 06 22:43:48 goodrobots-maverick systemd[1]: Failed to start LSB: OpenLDA....
Jul 06 22:43:48 goodrobots-maverick systemd[1]: slapd.service: Unit entered ....
Jul 06 22:43:48 goodrobots-maverick systemd[1]: slapd.service: Failed with r....
Hint: Some lines were ellipsized, use -l to show in full.
dpkg: error processing package slapd (--configure):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for libc-bin (2.23-0ubuntu10) ...
Errors were encountered while processing:
 slapd
E: Sub-process /usr/bin/dpkg returned an error code (1)
Error: /Stage[main]/Openldap::Server::Install/Package[slapd]/ensure: change from 'absent' to 'present' failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install slapd' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
slapd is already the newest version (2.4.42+dfsg-2ubuntu3.3).
The following packages were automatically installed and are no longer required:
  grub-pc-bin libllvm5.0 libproj-dev libvtk-java libvtk5.10
  linux-headers-4.4.0-109 linux-headers-4.4.0-109-generic
  linux-headers-4.4.0-112 linux-headers-4.4.0-112-generic
  linux-headers-4.4.0-116 linux-headers-4.4.0-116-generic
  linux-headers-4.4.0-124 linux-headers-4.4.0-124-generic
  linux-image-4.4.0-109-generic linux-image-4.4.0-112-generic
  linux-image-4.4.0-116-generic linux-image-4.4.0-124-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up slapd (2.4.42+dfsg-2ubuntu3.3) ...
  Moving old database directory to /var/backups:
  - directory unknown... done.
  Creating initial configuration... done.
  Creating LDAP directory... done.
Job for slapd.service failed because the control process exited with error code. See "systemctl status slapd.service" and "journalctl -xe" for details.
invoke-rc.d: initscript slapd, action "start" failed.
* slapd.service - LSB: OpenLDAP standalone server (Lightweight Directory Access Protocol)
   Loaded: loaded (/etc/init.d/slapd; bad; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2018-07-06 22:43:48 BST; 8ms ago
     Docs: man:systemd-sysv-generator(8)
  Process: 31496 ExecStart=/etc/init.d/slapd start (code=exited, status=1/FAILURE)

Jul 06 22:43:48 goodrobots-maverick slapd[31503]: ldif_read_file: Permission ...
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: DIGEST-MD5 common mech free
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: DIGEST-MD5 common mech free
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: slapd stopped.
Jul 06 22:43:48 goodrobots-maverick slapd[31503]: connections_destroy: nothin...
Jul 06 22:43:48 goodrobots-maverick slapd[31496]:    ...fail!
Jul 06 22:43:48 goodrobots-maverick systemd[1]: slapd.service: Control proce...1
Jul 06 22:43:48 goodrobots-maverick systemd[1]: Failed to start LSB: OpenLDA....
Jul 06 22:43:48 goodrobots-maverick systemd[1]: slapd.service: Unit entered ....
Jul 06 22:43:48 goodrobots-maverick systemd[1]: slapd.service: Failed with r....
Hint: Some lines were ellipsized, use -l to show in full.
dpkg: error processing package slapd (--configure):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for libc-bin (2.23-0ubuntu10) ...
Errors were encountered while processing:
 slapd
E: Sub-process /usr/bin/dpkg returned an error code (1)
Notice: /Stage[main]/Openldap::Server::Config/Shellvar[slapd]: Dependency Package[slapd] has failures: true
Warning: /Stage[main]/Openldap::Server::Config/Shellvar[slapd]: Skipping because of failed dependencies
Notice: /Stage[main]/Openldap::Server::Service/Service[slapd]: Dependency Package[slapd] has failures: true
Warning: /Stage[main]/Openldap::Server::Service/Service[slapd]: Skipping because of failed dependencies
Notice: /Stage[main]/Openldap::Server::Slapdconf/File[/srv/maverick/config/security/ldap]: Dependency Package[slapd] has failures: true
Warning: /Stage[main]/Openldap::Server::Slapdconf/File[/srv/maverick/config/security/ldap]: Skipping because of failed dependencies
Notice: /Stage[main]/Maverick_security::Ldap_server/File[/etc/ldap/slapd.d]: Dependency Package[slapd] has failures: true
Warning: /Stage[main]/Maverick_security::Ldap_server/File[/etc/ldap/slapd.d]: Skipping because of failed dependencies
Notice: /Stage[main]/Maverick_security::Ssl/Openssl::Dhparam[/srv/maverick/data/security/ssl/dhparam.pem]/File[/srv/maverick/data/security/ssl/dhparam.pem]/owner: owner changed 'mav' to 'root'
Notice: /Stage[main]/Maverick_security::Ssl/Openssl::Dhparam[/srv/maverick/data/security/ssl/dhparam.pem]/File[/srv/maverick/data/security/ssl/dhparam.pem]/group: group changed 'mav' to 'root'
Error: Failed to apply catalog: Execution of '/usr/sbin/slapcat -b cn=config -o ldif-wrap=no -H ldap:///???(objectClass=olcGlobal)' returned 1: 5b3fe29a could not stat config file "/etc/ldap/slapd.conf": No such file or directory (2)
slapcat: bad configuration file!
```

**Comment by fnoop** [comment]
Problem is slapd package is unusual in requiring a successful service start to install (which is really dumb, imho). 
`ldif_read_file: Permission denied for "/srv/maverick/config/security/ldap/cn=config.ldif"`

**Comment by fnoop** [comment]
Problem already hit in #689

**Comment by fnoop** [comment]
Raspberry 1.1.6 to 2.0-dev upgrading well now except for remaining slapd problem:
```
Error message: Execution of '/usr/bin/ldapmodify -Y EXTERNAL -H ldapi:/// -f /tmp/openldap_global_conf20180715-16438-1tc89n7' returned 80: SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
ldap_modify: Other (e.g., implementation specific) error (80)
modifying entry "cn=config"
Error: /Stage[main]/Openldap::Server::Slapdconf/Openldap::Server::Globalconf[TLSCACertificateFile]/Openldap_global_conf[TLSCACertificateFile]/ensure: change from 'absent' to 'present' failed: LDIF content:
dn: cn=config
add: olcTLSCACertificateFile
olcTLSCACertificateFile: /srv/maverick/data/security/ssl/ca/mavCA.pem

Error message: Execution of '/usr/bin/ldapmodify -Y EXTERNAL -H ldapi:/// -f /tmp/openldap_global_conf20180715-16438-1tc89n7' returned 80: SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
ldap_modify: Other (e.g., implementation specific) error (80)
modifying entry "cn=config"
```

**Comment by fnoop** [comment]
Fixed remaining ldap problem in #789
